libcloud/dns/drivers/digitalocean.py

```diff
@@ -0,0 +1,291 @@
+        """
+        Create a new record.
+
+        :param name: Record name without the domain name (e.g. www).
+                     Note: If you want to create a record for a base domain
+                     name, you should specify empty string ('') for this
+                     argument.
+        :type  name: ``str``
+
+        :param zone: Zone where the requested record is created.
+        :type  zone: :class:`Zone`
+
+        :param type: DNS record type (A, AAAA, ...).
+        :type  type: :class:`RecordType`
+
+        :param data: Data for the record (depends on the record type).
+        :type  data: ``str``
+
+        :param extra: Extra attributes for MX and SRV. (Depends on record)
+                      {"priority" : 0, "port" : 443, "weight" : 100}
+        :type extra: ``dict``
+
+        :rtype: :class:`Record`
+        """
+        params = {
+            "type": self.RECORD_TYPE_MAP[type],
+            "name": name,
+            "data": data
+        }
+        if extra:
+            try:
+                params['priority'] = extra['priority']
+            except KeyError:
+                params['priority'] = 'null'
+            try:
+                params['port'] = extra['port']
+            except KeyError:
+                params['port'] = 'null'
+            try:
+                params['weight'] = extra['weight']
+            except KeyError:
+                params['weight'] = 'null'
+
+        res = self.connection.request('/v2/domains/%s/records' % zone.id,
+                                      params=params,
+                                      method='POST')
+
+        return Record(id=res.object['domain_record']['id'],
+                      name=res.object['domain_record']['name'],
+                      type=type, data=data, zone=zone,
+                      driver=self, extra=extra)
+
+    def update_record(self, record, name=None, type=None,
+                      data=None, extra=None):
+        """
+        Update an existing record.
+
+        :param record: Record to update.
+        :type  record: :class:`Record`
+
+        :param name: Record name without the domain name (e.g. www). (Ignored)
+                     Note: The value is pulled from the record being updated
+        :type  name: ``str``
+
+        :param type: DNS record type (A, AAAA, ...). (Ignored)
+                     Note: Updating records does not support changing type
+                     so this value is ignored
+        :type  type: :class:`RecordType`
+
+        :param data: Data for the record (depends on the record type).
+        :type  data: ``str``
+
+        :param extra: (optional) Extra attributes (driver specific).
+        :type  extra: ``dict``
+
+        :rtype: :class:`Record`
+        """
+        params = {
+            "type": record.type,
+            "name": record.name,
+            "data": data
+        }
+        if data is None:
+            params['data'] = record.data
+        if extra:
```

Comment:
Since the same couple of lines of code are also used above this could perhaps be refactored into `def _get_params_from_extra_args` or a similar utility method.